cartoframes/context.py

```diff
@@ -169,14 +182,20 @@ def write(self, df, table_name, temp_dir='/tmp', overwrite=False,
             DataFarme has more than 100,000 rows, a :obj:`BatchJobStatus`
             instance is returned. Otherwise, None.
         """
-        if encode_geom:
-            _add_encoded_geom(df, geom_col)
+
+        pgcolnames = normalize_colnames(df.columns)
 
         if not overwrite:
             # error if table exists and user does not want to overwrite
             self._table_exists(table_name)
 
-        pgcolnames = normalize_colnames(df.columns)
+        if encode_geom:
+            if not HAS_GEOPANDAS:
+                raise RuntimeError('geopandas and shapely needs to be installed to use this option')
```

Comment:
Be explicit about 'this option'. maybe update to `encode_geom`

Reply:
Do you mean updating the docstring about encode_geom?

Reply:
When this Error is reported, the user may not know it's origins. So I think we should be specific by saying something like "`geopandas` and `shapely` need to be installed to use `encode_geom=True`"